### PR TITLE
[22851] Table Inline Create

### PR DESCRIPTION
**Pull request by oliverguenther**
Adds a new row for allowed users to create a work package within the table view.

Steps / TODOS
- [x] POST empty /api/v3/work_packages/form with first type for empty resource
- [x] Add the resulting wp resource to rows
- [x] Activate inline edit for all required fields
- [x] Focus into first required field
- [x] When pressing <enter>, create a new line starting with the above.

_Out of scope_ (handled in separate WP).
- [ ] Use Global post even in project context.
- [ ] Handle empty tables

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://oliverguenther-feature-inline-create-row.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-feature-inline-create-row.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

**Comment by oliverguenther** [comment]
https://community.openproject.com/work_packages/22851/activity

**Comment by HDinger** [comment]
I opened a PR which places the create button in the footer of the WP table. 
https://github.com/oliverguenther/openproject/pull/4

File: frontend/app/components/wp-buttons/wp-create-button/wp-create-button.directive.ts [hunk 1]
```diff
@@ -42,31 +40,8 @@ function wpCreateButton() {
...
-    ProjectService.fetchProjectResource(vm.projectIdentifier).then(function(project) {
-      canCreate = !!project.links.createWorkPackage;
-    });
-
-    ProjectService.getProject(vm.projectIdentifier).then(function (project) {
-      vm.types = project.embedded.types;
-    });
-  }
-}
+wpButtonsModule.directive('wpCreateButton', wpCreateButton);
```
Comment (furinvader) [inline frontend/app/components/wp-buttons/wp-create-button/wp-create-button.directive.ts:47]: @romanroe Here is an example, where we make use of the exported modules.

**Comment by furinvader** [comment]
Closed, as it is incorporated into #4290

**Pull request merged**